service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/OragnizationAgentMapperTest.java

```diff
@@ -0,0 +1,91 @@
+    private static final String INPUT_JSON_WITH_IDENTIFIER = ORGANIZATION_JSON + "1.json";
+    private static final String OUTPUT_XML_WITH_IDENTIFIER = ORGANIZATION_XML + "1.xml";
+    private static final String INPUT_JSON_WITHOUT_IDENTIFIER = ORGANIZATION_JSON + "2.json";
+    private static final String OUTPUT_XML_WITHOUT_IDENTIFIER = ORGANIZATION_XML + "2.xml";
+    private static final String INPUT_JSON_WITH_TELECOM_IN_0 = ORGANIZATION_JSON + "3.json";
+    private static final String OUTPUT_XML_WITH_TELECOM = ORGANIZATION_XML + "3.xml";
+    private static final String INPUT_JSON_WITHOUT_TELECOM = ORGANIZATION_JSON + "4.json";
+    private static final String OUTPUT_XML_WITHOUT_TELECOM = ORGANIZATION_XML + "4.xml";
+    private static final String INPUT_JSON_WITH_TELECOM_IN_2 = ORGANIZATION_JSON + "5.json";
+    private static final String OUTPUT_XML_WITH_TELECOM_2 = ORGANIZATION_XML + "5.xml";
+    private static final String INPUT_JSON_WITH_ADDRESS = ORGANIZATION_JSON + "6.json";
+    private static final String OUTPUT_XML_WITH_ADDRESS = ORGANIZATION_XML + "6.xml";
+    private static final String INPUT_JSON_WITHOUT_ADDRESS = ORGANIZATION_JSON + "7.json";
+    private static final String OUTPUT_XML_WITHOUT_ADDRESS = ORGANIZATION_XML + "7.xml";
+    private static final String INPUT_JSON_WITH_USE_HOME = ORGANIZATION_JSON + "8.json";
+    private static final String OUTPUT_XML_USE_HOME = ORGANIZATION_XML + "8.xml";
+    private static final String INPUT_JSON_WITH_USE_TEMP = ORGANIZATION_JSON + "9.json";
+    private static final String OUTPUT_XML_USE_TEMP = ORGANIZATION_XML + "9.xml";
+    private static final String INPUT_JSON_WITH_USE_EMPTY = ORGANIZATION_JSON + "10.json";
+    private static final String OUTPUT_XML_USE_EMPTY = ORGANIZATION_XML + "10.xml";
+
+    private OrganizationToAgentMapper organizationToAgentMapper;
+    @Mock
+    private RandomIdGeneratorService randomIdGeneratorService;
+
+    @BeforeEach
+    public void setUp() {
+        when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID);
+        MessageContext messageContext = new MessageContext(randomIdGeneratorService);
+        organizationToAgentMapper = new OrganizationToAgentMapper(messageContext);
+    }
+
+    @ParameterizedTest
+    @MethodSource("testArguments")
+    public void When_MappingOrganization_Expect_AgentResourceXml(String inputJson, String outputXml) throws IOException {
+        var jsonInput = ResourceTestFileUtils.getFileContent(inputJson);
+        var expectedOutput = ResourceTestFileUtils.getFileContent(outputXml);
+
+        Organization organization = new FhirParseService().parseResource(jsonInput, Organization.class);
+        var outputMessage = organizationToAgentMapper.mapOrganizationToAgent(organization);
+        System.out.println(outputMessage);
+        assertThat(outputMessage).isEqualToIgnoringWhitespace(expectedOutput);
+    }
+
+    private static Stream<Arguments> testArguments() {
+        return Stream.of(
```

Comment:
Nide to have/idea:
Did you consider reading all pairs of xml/json files from a given directory instead of explicitly defining their names upfront?
E.g.:
Define xml and json files as follows:
organization_with_identifier.json/organization_with_identifier.xml
organization_with_telecom.json/organization_with_telecom.xml
...

This way if we wanted to add new scenario we would only have to add json and xml files to resources directory.